name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/viewers/ParameterizedColumnLabelProvider.java

```diff
@@ -0,0 +1,25 @@
+package name.abuchen.portfolio.ui.util.viewers;
+
+import org.eclipse.jface.viewers.ColumnLabelProvider;
+import org.eclipse.swt.widgets.TableColumn;
+
+/**
+ * ColumnLabelProvider subclass which allows to access "option value"
+ * associated with a column. This can be e.g. reporting period (and
+ * there can be multiple instance of the same base column with different
+ * reporting periods set).
+ */
+public class ParameterizedColumnLabelProvider extends ColumnLabelProvider
+{
+    private TableColumn tableColumn;
+
+    public void setTableColumn(TableColumn tableColumn)
+    {
+        this.tableColumn = tableColumn;
```

Comment:
**That** is a nice idea. It will simplify. I just wonder if we should throw an exception if the tableColumn is already set. Then developers get an obvious hint if they re-use label provider (instead of using the factory method).

Anyway, will merge when I am back at my development machine.